File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/views/view-course-payment.js
// Framework and Technology Stack: C# Entity Framework

import CoursePayment from '../entities/course-payment';

class ViewCoursePayment extends CoursePayment {
    constructor() {
        super();
        this.Username = null;
        this.FullName = null;
        this.Avatar = null;
        this.Cover = null;
        this.Title = null;
        this.Thumbnail = null;
    }

    init() {
        return new ViewCoursePayment();
    }
}

export default ViewCoursePayment;

File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Views/ViewCoursePayment.cs
using KnowledgeSharingApi.Domains.Models.Entities.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Views
{
    [Table("ViewCoursePayment")]
    public class ViewCoursePayment : CoursePayment
    {
        public string FullName { get; set; } = string.Empty;

        public string? Avatar { get; set; }

        public string? Cover { get; set; }

        public string Title { get; set; } = string.Empty;

        public string? Thumbnail { get; set; }

        protected override ViewCoursePayment Init()
        {
            return new ViewCoursePayment();
        }
    }
}
